#!/usr/bin/env python3
PACKAGE = "controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('kpX', double_t, 0, "kpX", 0.14, 0.0, 2.0)
gen.add('kpY', double_t, 0, "kpY", 0.2, 0.0, 2.0)
gen.add('kdX', double_t, 0, "kdX", 0.016, 0.0, 2.0)
gen.add('kdY', double_t, 0, "kdY", 0.04, 0.0, 2.0)
gen.add('kpT', double_t, 0, "kpT", 0.01, 0.0, 2.0)
gen.add('kdT', double_t, 0, "kdT", 0.008, 0.0, 2.0)

gen.add('xDes', double_t, 0, "xDes", 0.0,  -10.0, 10.0)
gen.add('yDes', double_t, 0, "yDes", 0.0,  -10.0, 10.0)
gen.add('tDes', double_t, 0, "tDes", 0.0,  -180, 180)

gen.add('h', double_t, 0, "h", 0.2,  0.01, 2.0)

exit(gen.generate(PACKAGE, "controller", "cf_dina"))
